src/swaps/liqualityboost/LiqualityBoostSwapProvider.js

```diff
@@ -0,0 +1,290 @@
+    const fromClient = this.getClient(network, walletId, quote.from, quote.fromAccountId)
+
+    const message = [
+      'Creating a swap with following terms:',
+      `Send: ${quote.fromAmount} (lowest denomination) ${quote.from}`,
+      `Receive: ${quote.toAmount} (lowest denomination) ${quote.to}`,
+      `My ${quote.from} Address: ${quote.fromAddress}`,
+      `My ${quote.to} Address: ${quote.toAddress}`,
+      `Counterparty ${quote.from} Address: ${quote.fromCounterPartyAddress}`,
+      `Counterparty ${quote.to} Address: ${quote.toCounterPartyAddress}`,
+      `Timestamp: ${quote.swapExpiration}`
+    ].join('\n')
+
+    const messageHex = Buffer.from(message, 'utf8').toString('hex')
+    const secret = await fromClient.swap.generateSecret(messageHex)
+    const secretHash = sha256(secret)
+
+    const fromFundTx = await fromClient.swap.initiateSwap(
+      {
+        value: BN(quote.fromAmount),
+        recipientAddress: quote.fromCounterPartyAddress,
+        refundAddress: quote.fromAddress,
+        secretHash: secretHash,
+        expiration: quote.swapExpiration
+      },
+      quote.fee
+    )
+
+    return {
+      ...quote,
+      status: 'INITIATED',
+      secret,
+      secretHash,
+      fromFundHash: fromFundTx.hash,
+      fromFundTx,
+      slippage: slippagePercentage * 100
+    }
+  }
+
+  async estimateFees ({ network, walletId, asset, txType, quote, feePrices, max }) {
+    const liqualityFees = await this.liqualitySwapProvider.estimateFees({ network, walletId, asset, txType, quote: { ...quote, to: quote.toNativeAsset, toAmount: quote.toNativeAssetAmount }, feePrices, max })
+    if (isERC20(asset) && txType === LiqualityBoostSwapProvider.txTypes.SWAP_CLAIM) {
+      const oneinchFees = await this.oneinchSwapProvider.estimateFees({ network, walletId, asset, txType: LiqualityBoostSwapProvider.txTypes.SWAP, quote: { ...quote, from: quote.toNativeAsset, fromAmount: quote.toNativeAssetAmount, fromAccountId: quote.toAccountId, slippagePercentage }, feePrices, max })
+      const totalFees = {}
+      for (const key in oneinchFees) {
+        totalFees[key] = BN(oneinchFees[key]).plus(liqualityFees[key])
+      }
+      return totalFees
+    }
+    return liqualityFees
+  }
+
+  async finalizeLiqualitySwapAndStartOneinch ({ swap, network, walletId }) {
+    const result = await this.liqualitySwapProvider.waitForClaimConfirmations({ swap, network, walletId })
+    if (result?.status === 'SUCCESS') return { endTime: Date.now(), status: 'SEND_ONEINCH_SWAP' }
+  }
+
+  async performNextSwapAction (store, { network, walletId, swap }) {
+    let updates
+    const swapLiqualityFormat = { ...swap, to: swap.toNativeAsset, toAmount: swap.toNativeAssetAmount, slippagePercentage }
+    const swapOneInchFormat = { ...swap, from: swap.toNativeAsset, fromAmount: swap.toNativeAssetAmount, fromAccountId: swap.toAccountId, slippagePercentage }
```

Comment:
Similar to other comment is it possible to do liek this?
```
let updates = this.liqualitySwapProvider.performNextAction({swap: swapLiqualityFormat})
if (!updates) {
updates = this.oneinchSwapProvider.performNextAction({swap: swapOneInchFormat})
}
return updates
```

Reply:
could also take care of the success check in `finalizeLiqualitySwapAndStartOneinch` in the middle of the two

Reply:
> Similar to other comment is it possible to do liek this?
> 
> ```
> let updates = this.liqualitySwapProvider.performNextAction({swap: swapLiqualityFormat})
> if (!updates) {
> updates = this.oneinchSwapProvider.performNextAction({swap: swapOneInchFormat})
> }
> return updates
> ```

Makes sense. I will try to think how to add more abstraction in order to implement this.